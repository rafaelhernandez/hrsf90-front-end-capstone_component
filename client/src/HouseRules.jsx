import React from 'react';
import { Glyphicon } from 'react-bootstrap';

const HouseRules = (props) => (
  <div className="div-description-house-rules">
    <div className="div-description-checkin-time">
      Check-in is anytime after {props.desc.localized_check_in_time_window}
    </div>
    <div className="div-description-checkout-time">
      Check out by {props.desc.localized_check_out_time}
    </div>
    <div className="div-description-read-rules-fold">
      <a href="#">Read all rules <Glyphicon glyph="chevron-down"/></a>
    </div>
    <div className="div-description-additional-house-rules">
      {props.desc.additional_house_rules}
    </div>
    <div className="div-description-house-rules-acknowledge">
      You must also acknowledge
    </div>
    <div className="div-description-deposit">
      {props.desc.listing_security_deposit_native > 0 ?
        'Security Deposit - if you damage the home, you may be charged up to ' + props.desc.security_deposit_formatted :
        ''
      }
    </div>
    <div className="div-description-hide-rules-fold">
      <a href="#">Hide rules <Glyphicon glyph="chevron-up"/></a>
    </div>
  </div>
);

export default HouseRules;
