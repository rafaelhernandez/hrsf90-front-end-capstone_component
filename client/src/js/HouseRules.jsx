import React from 'react';
import { Glyphicon } from 'react-bootstrap';

const HouseRules = (props) => (
  <div className="div-description-house-rules">
    <div className="div-description-section-header">
      House Rules
    </div>
    <div className="div-description-house-rules-details">
      <div className="div-description-checkin-time">
        Check-in is {props.desc.localized_check_in_time_window}
      </div>
      <div className="div-description-checkout-time">
        Check out by {props.desc.localized_check_out_time}
      </div>
      <div className="div-description-read-rules-fold">
        <p className="div-description-separator-margin">
          <a className="a-description-fold" href="#">Read all rules <Glyphicon glyph="chevron-down"/></a>
        </p>
      </div>
      <div className="div-description-additional-house-rules">
        {props.desc.additional_house_rules}
      </div>
      <div className="div-description-house-rules-acknowledge">
        You must also acknowledge
      </div>
      <div className="div-description-deposit">
        {props.desc.listing_security_deposit_native > 0 ?
          'Security Deposit - if you damage the home, you may be charged up to ' + props.desc.security_deposit_formatted :
          ''
        }
      </div>
      <div className="div-description-hide-rules-fold">
        <p className="div-description-separator-margin">
          <a className="a-description-fold" href="#">Hide rules <Glyphicon glyph="chevron-up"/></a>
        </p>
      </div>
    </div>
  </div>
);

export default HouseRules;
