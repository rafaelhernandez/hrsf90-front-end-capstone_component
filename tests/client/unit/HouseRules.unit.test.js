import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import HouseRules from '../../../client/src/js/HouseRules.jsx';
import descriptionSample from '../../../db/data/description_sample.js';

configure({ adapter: new Adapter() });

describe('HouseRules', () => {
  it('should render correctly', () => {
    const output = shallow(
      <HouseRules desc={descriptionSample}></HouseRules>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
