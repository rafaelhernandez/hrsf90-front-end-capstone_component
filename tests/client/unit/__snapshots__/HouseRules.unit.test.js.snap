// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HouseRules should render correctly 1`] = `
<div
  className="div-description-house-rules"
>
  <div
    className="div-description-section-header"
  >
    House Rules
  </div>
  <div
    className="div-description-house-rules-details"
  >
    <div
      className="div-description-checkin-time"
    >
      Check-in is 
      Anytime after 2PM
    </div>
    <div
      className="div-description-checkout-time"
    >
      Check out by 
      12PM (noon)
    </div>
    <div
      className="div-description-read-rules-fold"
    >
      <p
        className="div-description-separator-margin"
      >
        <a
          className="a-description-fold"
          href="#"
        >
          Read all rules 
          <Glyphicon
            bsClass="glyphicon"
            glyph="chevron-down"
          />
        </a>
      </p>
    </div>
    <div
      className="div-description-additional-house-rules"
    >
      I'm OK with pets and the unit opens up to the backyard which makes it easy but my dog is out there so your pet must get along!
    </div>
    <div
      className="div-description-house-rules-acknowledge"
    >
      You must also acknowledge
    </div>
    <div
      className="div-description-deposit"
    >
      Security Deposit - if you damage the home, you may be charged up to $250
    </div>
    <div
      className="div-description-hide-rules-fold"
    >
      <p
        className="div-description-separator-margin"
      >
        <a
          className="a-description-fold"
          href="#"
        >
          Hide rules 
          <Glyphicon
            bsClass="glyphicon"
            glyph="chevron-up"
          />
        </a>
      </p>
    </div>
  </div>
</div>
`;
